### ISSUE TITLE
ES6 modules broken in 0.10.8 with Salsa

### ISSUE THREAD

**Issue by egamma** [Microsoft/TypeScript#7006]
_From @prashaantt on February 9, 2016 15:34_

I'm loving the JSX support in the latest Code update, courtesy Salsa — thanks a ton for that! The thing is, I'm not ready to go all out on TS yet. Is it possible to just get the JSX auto-complete/IntelliSense support without running into issues like these:

![image](https://cloud.githubusercontent.com/assets/390540/12920869/7e7322ba-cf70-11e5-9768-cb8768e1d13f.png)

![image](https://cloud.githubusercontent.com/assets/390540/12920907/a92d7276-cf70-11e5-85b9-c1e8328d37c2.png)

_Copied from original issue: Microsoft/vscode#2848_

**Comment by egamma** [comment]
_From @prashaantt on February 9, 2016 16:1_

~~I have been experimenting a bit and dropping a `jsconfig.json` in my project seems to do the trick and no longer shows these errors. Is that the way to achieve this?~~

I jumped too early. A `jsconfig.json` fixes the first error with the import but not the second one with the export.

**Comment by egamma** [comment]
_From @tmkbry on February 9, 2016 16:20_

@prashaantt I had the same problem, the jsconfig.json fixed it for me, thanks.

To the second problem, It doesn't show an error for me, maybe it's because I have this in my jsconfig.json:
`{
    "compilerOptions": {
        "target": "ES6"
    },
    "exclude": [
        "node_modules"
    ]
}`

**Comment by egamma** [comment]
_From @prashaantt on February 9, 2016 16:29_

@tmkbry I have a similar config as yours, plus the `"module": "commonjs"` inside `compilerOptions`, but I can't get rid of the squigglies on the export either way.

EDIT: I think ideally the module should be `es6` but `jsconfig.json` reports an error for that, which otherwise is acceptable inside a `tsconfig.json`.

**Comment by egamma** [comment]
_From @prashaantt on February 9, 2016 18:35_

Ok, so this works:

```js
export default function () { }
```

but this doesn't:

```js
export default () => { }
```

Awesome!

**Comment by egamma** [comment]
_From @prashaantt on February 9, 2016 18:37_

(Perhaps I should leave this issue open, just in case there's anything the Code team can do to make this alt syntax work too.)

**Comment by egamma** [comment]
_From @prashaantt on February 9, 2016 19:26_

Another instance that I can't work around:

```js
import { connect } from 'react-redux';

const mapStateToProps = (state) => { ... };

class App extends React.Component { ... }

export default connect(mapStateToProps)(App); // here be the relentless squiggly
```

The `export` throws the same error as before: `'export=' can only be used in a .ts file`.

**Comment by egamma** [comment]
Extracted a separate issue for 
>I think ideally the module should be  es6  but  jsconfig.json  reports an error for that, which otherwise is acceptable inside a  tsconfig.json .

Moving to Salsa

**Comment by billti** [comment]
There seems to be a few issues above. What is the specific problem here? Is it that ES6 should be the default module type? (which would be incompatible with the other request for CommonJS to be the default modules type)? Or is that `export default` doesn't work for arrow functions?

**Comment by prashaantt** [comment]
It's more like `export default` doesn't work for some cases.

Here are a few things that don't work:

```js
const a = function () {}; // this could be whatever: object or function, arrow or not, anonymous or named

export default a;
```

```js
function myFunc() {};

export default myFunc();
```

---

These *do* work:

```js
export const a = function () {}; // could be anything
```

```js
export default function myFunc() {}; // anonymous is also ok
```

```js
const a = function () {};

export { a };
```

```js
const a = function () {};

module.exports = a;
```

**Comment by billti** [comment]
I just tried all the things above which state they **don't** work, using the latest `release-1.8` build, and they all look fine for me with no errors (including the React/Redux pattern above). Can you try with the latest bits and see if this is still an issue? Else looks like we've addressed the above patterns.

Thanks!

<img width="420" alt="screen shot 2016-02-12 at 10 50 59 am" src="https://cloud.githubusercontent.com/assets/993909/13017218/f277d51c-d177-11e5-932a-63d11b268e8b.png">

<img width="298" alt="screen shot 2016-02-12 at 10 51 26 am" src="https://cloud.githubusercontent.com/assets/993909/13017221/f9086090-d177-11e5-881d-c84f1487bd83.png">

**Comment by prashaantt** [comment]
@billti Thanks for looking into this.

As recommended in the [release notes](https://code.visualstudio.com/Updates#_javascript-salsa-preview) for VS Code, my error report was based on `typescript@1.9.0-dev.20160128`.

Per your request, I tried moving to the latest 1.8 release version on npm `1.8.0` and still see these errors. I then switched to `typescript@next` which currently gives me `1.9.0-dev.20160212` where all these errors indeed don't show up, so all looks good there.

But since you specifically asked me to check on `1.8` I'm wondering if its latest version is up on npm at all, because the currently `1.8.0` tagged version (2016-01-28T19:54:42.196Z) didn't fix these issues for me.

**Comment by billti** [comment]
@prashaantt typescript@next is whatever is currently in the `master` branch, so glad to see it's fixed there too! Thanks for checking.

`release-1.8` is the release we're about to ship. Once it forks from `master`, then it only gets a new release on NPM when we push it, so it can be a little behind (as it is currently). There should be a new (hopefully final 1.8) up there shortly though.

**Comment by prashaantt** [comment]
Awesome, thanks!
